gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/crud_service/access_point/AccessPointCrudServiceImpl.java

```diff
@@ -70,11 +94,23 @@ public void deleteAccessPoints(final AccessPoint.ReferenceType referenceType, fi
                 AccessPointReferenceType.valueOf(referenceType.name()),
                 referenceId
             );
+
             if (deleteAccessPoints != null) {
-                deleteAccessPoints
-                    .stream()
-                    .map(AccessPointAdapter.INSTANCE::toEntity)
-                    .forEach(accessPoint -> eventManager.publishEvent(AccessPointEvent.DELETED, accessPoint));
+                for (io.gravitee.repository.management.model.AccessPoint accessPoint : deleteAccessPoints) {
+                    eventManager.publishEvent(AccessPointEvent.DELETED, AccessPointAdapter.INSTANCE.toEntity(accessPoint));
```

Comment:
Not sure the purpose of this event?

Reply:
Other comment for an update event makes sense and I will make those changes. I could keep this event just for when this delete is called from somewhere other than the updateAccessPoints method (so cleanup such as unlinking and disabling environments), or just remove it and not clean up access points for unlinked/disabled environments. 

I'm thinking to keep the delete just for other cases with a parameter on the method to not publish the event if delete is from update. Does that make sense/do you agree @guillaumelamirand ?

Reply:
I mean the event will be propagated on the mAPI event manager and I don't see anyone using it.

Reply:
ah the event manager one rather than the new one, I'll have a look